src/cpp/src/utils.cpp

```diff
@@ -292,10 +292,27 @@ bool is_gguf_model(const std::filesystem::path& file_path) {
 
 } // namespace
 
-std::shared_ptr<ov::Model> read_model(const std::filesystem::path& model_dir,  const ov::AnyMap& config) {
+std::shared_ptr<ov::Model> read_model(const std::filesystem::path& model_dir,  const ov::AnyMap& properties) {
     if (is_gguf_model(model_dir)) {
 #ifdef ENABLE_GGUF
-        return create_from_gguf(model_dir.string());
+        if (properties.find(ov::cache_dir.name()) != properties.end()) {
+            std::string cache_dir = properties.at(ov::cache_dir.name()).as<std::string>();
+            if (!cache_dir.empty()) {
+                std::filesystem::path model_cache_dir(cache_dir);
+                std::filesystem::path model_path = model_cache_dir / (model_dir.stem().string() + "_openvino_model.xml");
+                if (std::filesystem::exists(model_path)) {
+                    std::cout << "Found generated OpenVINO model: " << model_path.string() << ", skip creating from GGUF model.\n";
```

Comment:
After the model is saved, the next `LLMPipeline()` constructor can read the folder as usual ignoring GGUF files. That's not easy to implement in a sample to make everything short, readable and explicit, but in general case users can trigger GGUF branch for the fist constructor and OV branch for the next constructors. That would shift cache invalidation responsibility to the user. Would that work for you?

Reply:
Hi @Wovchena, thanks for your feedback. May I double check that your suggest not to use `cache_dir` property to control whether save OV model explictly, instead we implicit save OV model for first run (GGUF branch), and re-use OV model for 2nd run (OV branch). Right?

Reply:
Correct. This is also possible flip the implicitness: GGUF doesn't save the model by default, but if the a special property is given, the model is saved. Implicitness or explicitness doesn't affect the general idea.

Reply:
There also should be a way to disable model saving. For example if someone doesn't have enough space or GGUF model is on a share mounted as read only

Reply:
Thanks, I've update my implementation to address your comments.